### PR TITLE
Refine arm testlists and save more jobs for later usage

### PR DESCRIPTION
**Pull request by liuguo09**
1. Refine arm testlists with 35 configs in average
2. Remove arm-14 and arm-15 jobs to save more jobs for Windows and MacOS
builds lately since github action free version limit up to 20 jobs.

Signed-off-by: liuhaitao <liuhaitao@xiaomi.com>

### CODE REVIEW

**Comment by yamt** [comment]
> github action free version limit up to 20 jobs.

i wasn't aware of the limit. where this kind of things are documented?

**Comment by liuguo09** [comment]
> > github action free version limit up to 20 jobs.
> 
> i wasn't aware of the limit. where this kind of things are documented?

https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits

**Comment by yamt** [comment]
thank you.

do this Storage/Minutes limits apply as well?
https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions#about-billing-for-github-actions

**Comment by btashton** [comment]
I don't think we should worry too much. If we have more than 20 jobs they will just wait for others to free up. The startup time is fairly low now and there is still room for improvement by limiting the depth of the clones we do. 

Also make distclean is kinda slow. We can be using git -xdf to reset state a lot faster.

**Comment by yamt** [comment]
the change looks good to me. (as far as it's merged to all repos at once)
it covers the same set of configs as before.

File: testlist/arm-01.dat [hunk 1]
```diff
@@ -6,6 +6,6 @@
 
 /arm/cxd56xx,CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
 
-/arm/dm320,CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
-
 /arm/efm32,CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
+
+/arm/stm32f0l0g0,CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
```
Comment (xiaoxiang781216) [inline testlist/arm-01.dat:11]: Could we order the list? so we can find the untested board quickly.

Comment (liuguo09) [reply testlist/arm-01.dat:11]: If in the alphabetic order completely, we should use more level depth boards path instead.

Comment (liuguo09) [reply testlist/arm-01.dat]: Now refine in  alphabetic order done.

**Pull request merged**